src/flb_upstream.c

```diff
@@ -838,16 +839,21 @@ int flb_upstream_conn_timeouts(struct mk_list *list)
                     }
                 }
 
+                inject = FLB_FALSE;
                 if (u_conn->event.status != MK_EVENT_NONE) {
-                    mk_event_inject(u_conn->evl,
-                                    &u_conn->event,
-                                    u_conn->event.mask,
-                                    FLB_TRUE);
+                    inject = FLB_TRUE;
                 }
-
                 u_conn->net_error = ETIMEDOUT;
-
                 prepare_destroy_conn(u_conn);
+                /* 
+                 * prepare_destroy_conn calls mk_list_del on the event in the 
+                 * priority bucket queue, so for safety, we inject it afterwards
+                 */
+                if (inject == FLB_TRUE) {
+                    mk_event_inject(u_conn->evl, &u_conn->event,
+                                    MK_EVENT_READ | MK_EVENT_WRITE,
```

Comment:
This change would introduce a regression, please use  `u_conn->event.mask` as the original code did.

Reply:
oops

Reply:
fixed